### ISSUE TITLE
Crouton updates/setup failing after module reinstalls for VirtualBox

### ISSUE THREAD

**Issue by laurenweinstein1** [dnschneid/crouton#2364]
Hi. As per the discussion in: https://github.com/divx118/crouton-packages/issues/27 I needed to use the technique in https://github.com/dnschneid/crouton/wiki/Build-kernel-headers-and-install-Virtualbox-(x86) to get VirtualBox running again (it had previously been running fine, originally installed via the same process). This ultimately succeeded. However, since the build and install of the new modules for 3.8.11, there are more basic problems.

Ubuntu updater keeps trying to download/install a kernel image for 3.8.11. This install always fails, so it keeps trying repeatedly.

On the CrOS side, my attempt to do a crouton-update triggered a similar failure, causing the crouton setup to fail (and showing again a failure related to a 3.8.11 image/modules). After that setup failure (I captured the original error text, below) I was able to continue the script -- there were more errors I couldn't capture and I ultimately got back into the chroot (which I do not want to risk being blocked from). I can restart again now but obviously I haven't tried another chroot-update and am ignoring the Ubuntu updater's attempts to download a 3.8.11 image. (I'm already now running 3.8.11 of course, and the modules are all in /lib/modules/3.8.11.

Error text from crouton-update is below. Any ideas to fix this configuration problem and avoid it in the future would be greatly appreciated. Thanks! 

---

Fetched 5121 B in 16s (302 B/s)

Reading package lists... Done
Ensuring system is up-to-date...
Reading package lists... Done
Building dependency tree

Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
wine
The following packages will be upgraded:
linux-image-3.8.11
1 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
Need to get 0 B/6860 kB of archives.
After this operation, 650 kB disk space will be freed.
Preconfiguring packages ...
(Reading database ... 216838 files and directories currently installed.)
Preparing to unpack .../linux-image-3.8.11_20150314_amd64.deb ...
Ok, aborting, since modules for this image already exist.
dpkg: error processing archive /var/cache/apt/archives/linux-image-3.8.11_20150314_amd64.deb (--unpack):
subprocess new pre-installation script returned error exit status 1
Examining /etc/kernel/postrm.d .
run-parts: executing /etc/kernel/postrm.d/initramfs-tools 3.8.11 /boot/vmlinuz-3.8.11
Errors were encountered while processing:
/var/cache/apt/archives/linux-image-3.8.11_20150314_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
Failed to complete chroot setup.

**Comment by divx118** [comment]
As said in the issue on my github, remove `/etc/apt/sources.list.d/crouton-packages.list` This will remove the source of my git repo. Then `sudo apt-get update && sudo apt-get upgrade` to get everything back to normal. After that your chroot should update just fine.

**Comment by laurenweinstein1** [comment]
Unfortunately -- due to the aborted setup script? -- CrOS shell is now reporting apt-get not found.

**Comment by nigelhealy** [comment]
On Tue, Jan 12, 2016 at 8:42 AM, Lauren Weinstein <notifications@github.com>
wrote:

> Hi As per the discussion in:
> https://githubcom/divx118/crouton-packages/issues/27 I needed to use the
> technique in
> https://githubcom/dnschneid/crouton/wiki/Build-kernel-headers-and-install-Virtualbox-(x86)
> to get VirtualBox running again (it had previously been running fine,
> originally installed via the same process) This ultimately succeeded
>

I also followed that guide successfully built the kernel debs and installed
but when I run virtualbox it asks to run setup which then panics and
reboots. Please can you focus on the steps I have done wrong, probably
towards the end?

**Comment by laurenweinstein1** [comment]
Bingo. Everything now appears back to normal, and crouton-update also worked. I have to admit I do not fully understand the intricacies of the CrOS/Crouton/chroot architecture. Thanks again!

**Comment by divx118** [comment]
@nigelhealy Yours is a different problem, @laurenweinstein1 is running 3.8.11 kernel which doesn't have a problem with loading modules yet. I am still looking trhough the 3.14 kernel commits see if I can find something there that could be related. Maybe @drinkcat or @dnschneid could take a look at [the kernel panic](https://gist.github.com/divx118/4bed590e37046ce1b201#file-kernel-panic-3-14-chromeos-virtualbox-modules) to see what is causing it when loading virtualbox modules on a 3.14 kernel.

**Comment by fthnhp** [comment]
Excuse me @divx118, is it still on process? I've already fucked crouton a few times. I actually thought the higher kernel the less messy. I should have read about my panic kernel before.:(

**Comment by divx118** [comment]
@fthnhp The kernel panic for 3.14.0 should be resolved on stable channel see https://github.com/divx118/crouton-packages/issues/26

**Comment by fthnhp** [comment]
@divx118 I don't have a pixel, but an acer c910
So, Is it safe to follow the old thread (install kernels)?

**Comment by divx118** [comment]
@fthnhp You can follow the easy way https://github.com/divx118/crouton-packages/blob/master/README.md or do it all by yourself the harder way and follow https://github.com/dnschneid/crouton/wiki/Build-kernel-headers-and-install-Virtualbox-(x86).
Both shouldn't be a problem. However that said it is always recommended to have backups from any important files you have on your chromebook.

**Comment by fthnhp** [comment]
Thanks sir @divx118, I really appreciate it .I've followed the easy way but I'm still getting an error. And I don't know how to solve it:
(Reading database ... 135814 files and directories currently installed.)
Preparing to unpack .../virtualbox_4.3.36-dfsg-1+deb8u1ubuntu1.14.04.1_amd64.deb ...
Unpacking virtualbox (4.3.36-dfsg-1+deb8u1ubuntu1.14.04.1) over (4.3.36-dfsg-1+deb8u1ubuntu1.14.04.1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up virtualbox (4.3.36-dfsg-1+deb8u1ubuntu1.14.04.1) ...
 * Stopping VirtualBox kernel modules                                                                     [ OK ] 
 * Starting VirtualBox kernel modules                                                                             * modprobe vboxdrv failed. Please use 'dmesg' to find out why
                                                                                                          [fail]
invoke-rc.d: initscript virtualbox, action "restart" failed.

When I type dmesg I get:
[ 3604.918939] tpm_tis tpm_tis: command 0x65 (size 20) returned code 0x0
[ 3604.948760] tpm_tis tpm_tis: command 0x65 (size 22) returned code 0x0
[ 3604.979141] tpm_tis tpm_tis: command 0x65 (size 22) returned code 0x0
[ 3608.964297] Chromium OS LSM: init_module denied obj="/lib/modules/3.14.0/updates/dkms/vboxdrv.ko" pid=16291 cmdline="modprobe vboxdrv"
[ 3724.723246] Chromium OS LSM: init_module denied obj="/lib/modules/3.14.0/updates/dkms/vboxdrv.ko" pid=17503 cmdline="modprobe vboxdrv"
[ 4029.815186] Chromium OS LSM: init_module denied obj="/lib/modules/3.14.0/updates/dkms/vboxdrv.ko" pid=19209 cmdline="modprobe vboxdrv"

And when I try modprobe vboxdrv
modprobe: ERROR: could not insert 'vboxdrv': Operation not permitted

Note that I solved an invoke.d: policy.d error
allowed with: echo exit 0 > /usr/sbin/policy-rc.d
and assured it's executable  chmod +x /usr/sbin/policy-rc.d

I wish I could install it because I'm going to have some virtualizations for web services practises.

**Comment by nigelhealy** [comment]
Your first problem looks like you didn't install the headers first.

Your 2nd problem looks like you didn't do as root, but then you shouldn't need to run it anyway if you used the script.

Run the setup-headers.sh 
then logout
then try running virtualbox. It may well have not correctly installed correctly because you installed before installing headers, so you do a sudo apt-get install --reinstall again.

Also, why are you install Virtualbox4.3 and not Virtualbox5.0 ?

**Comment by fthnhp** [comment]
When I run apt-get install I get 4.3 by default. But I really don't mind.
I run it as super user, so that wasn't the case.
I made a purge virtualbox and follow your instructions and now I'm at the beginning again :(

Setting up virtualbox (4.3.36-dfsg-1+deb8u1ubuntu1.14.04.1) ...
runlevel:/var/run/utmp: No such file or directory
 * Stopping VirtualBox kernel modules                                                                                                                                                                                         [ OK ] 
 * Starting VirtualBox kernel modules                                                                                                                                                                                                 * No suitable module for running kernel found
                                                                                                                                                                                                                              [fail]
invoke-rc.d: initscript virtualbox, action "restart" failed.

**Comment by fthnhp** [comment]
@nigelhealy Also, I've tried to build the kernel and there's no way
(ubuntu)alvaro@localhost:~$ sudo m-a prepare
Getting source for kernel version: 3.14.0
Kernel headers available in /usr/src/linux-headers-3.14.0
Creating symlink...
apt-get install build-essential 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

Done!
(ubuntu)alvaro@localhost:~$ sudo /etc/init.d/virtualbox restart
 * Stopping VirtualBox kernel modules                                                                                                                                                                                         [ OK ] 
 * Starting VirtualBox kernel modules                                                                                                                                                                                                 * No suitable module for running kernel found

**Comment by nigelhealy** [comment]
https://github.com/divx118/crouton-packages/blob/master/README.md

Did you in ChromeOS run change-kernel-flags and then reboot?

In your chroot, what happened when you 
sudo sh setup-headers.sh

Is your /etc/rc.local modified?

If you've done all those then the install of virtualbox failed then looks a bug but you're on the same kernel as me 3.14 and mine is fully working now.

**Comment by fthnhp** [comment]
@nigelhealy I think now I run it properly since I got no errors but warnings on terminal..
Just reinstalled virtualbox-dkms..
Now I'm downloading WIn10.. let's see haha

**Comment by fthnhp** [comment]
Excuse me, @divx118 I've already download a 32b win distro, and avoided exceed half of the cores and ram, due to it's a i3. But it's not working yet. When tries to run (after a popup) I get an error

VT-x is disabled in the BIOS. (VERR_VMX_MSR_VMXON_DISABLED).
Result Code: 
NS_ERROR_FAILURE (0x80004005)
Component: 
Console
Interface: 
IConsole {8ab7c520-2442-4b66-8d74-4ff1e195d2b6}

But I'm pretty sure I enabled VT-x a few months ago. Btw I've just tried this https://gist.github.com/DennisLfromGA/cd3455530cec2a5a1ef4, and nothing's new.
